event-sourcing/event-source-api-integration-test/src/test/java/uk/gov/justice/services/eventsourcing/source/api/EventsFeedIT.java

```diff
@@ -0,0 +1,377 @@
+
+        eventsRepository.insert(eventOf(1L, streamId));
+        eventsRepository.insert(eventOf(2L, streamId));
+
+        eventFeedService.initialiseWithPageSize(2);
+
+        final HttpResponse response = eventsFeedFor(SYSTEM_USER_ID, streamId);
+
+        with(responseBodyOf(response))
+                .assertNotDefined("$.paging.previous");
+
+    }
+
+    @Test
+    public void shouldNotPresentLinkToPreviousPageIfOn1stPage() throws Exception {
+
+        final UUID streamId = randomUUID();
+
+        eventsRepository.insert(eventOf(1L, streamId));
+        eventsRepository.insert(eventOf(2L, streamId));
+        eventsRepository.insert(eventOf(3L, streamId));
+
+        eventFeedService.initialiseWithPageSize(2);
+
+        final HttpResponse response = eventsFeedFor(SYSTEM_USER_ID, randomUUID());
+
+        final String responseBody = responseBodyOf(response);
+
+        with(responseBody)
+                .assertNotDefined("$.paging.previous");
+
+    }
+
+    @Test
+    public void shouldReturnForbiddenIfNotASystemUser() throws IOException {
+        final HttpResponse response = eventsFeedFor(randomUUID(), randomUUID());
+        assertThat(response.getStatusLine().getStatusCode(), is(FORBIDDEN.getStatusCode()));
+    }
+
+    private String responseBodyOf(final HttpResponse response) throws IOException {
+        return EntityUtils.toString(response.getEntity());
+    }
+
+    private HttpResponse eventsFeedFor(final UUID userId, final UUID streamId) throws IOException {
+        final String url = format(BASE_URI_PATTERN + "/event-streams/" + streamId.toString(), port);
+        return feedOf(url, userId);
+    }
+
+    private HttpResponse feedOf(final String url, final UUID userId) throws IOException {
+        final HttpUriRequest request = new HttpGet(url);
+        request.addHeader(USER_ID, userId.toString());
+        return httpClient.execute(request);
+    }
+
+    private void initEventDatabase() throws Exception {
+
+        Liquibase eventStoreLiquibase = new Liquibase(LIQUIBASE_EVENT_STORE_CHANGELOG_XML,
+                new ClassLoaderResourceAccessor(), new JdbcConnection(dataSource.getConnection()));
+        eventStoreLiquibase.dropAll();
+        eventStoreLiquibase.update("");
+    }
+
+    private Event eventOf(final long sequenceId, final UUID streamId) {
+        return new Event(randomUUID(), streamId, sequenceId, NAME, METADATA_JSON, PAYLOAD_JSON, TIMESTAMP);
+    }
+
+    @ApplicationScoped
+    public static class TestSystemUserProvider implements SystemUserProvider {
+
+        @Override
+        public Optional<UUID> getContextSystemUserId() {
+            return Optional.of(SYSTEM_USER_ID);
+        }
+    }
+
+
+    @ApplicationScoped
+    public static class TestEventInsertionStrategyProducer {
+
+        @Produces
+        public EventInsertionStrategy eventLogInsertionStrategy() {
+            return new AnsiSQLEventLogInsertionStrategy();
+        }
+    }
+
+}
```

Comment:
it would also be good to add a test case in the EventsFeedIT that follows link from eventStreams feed to the specific events feed